### PR TITLE
WIP: Setup whitelist

### PR DESCRIPTION
**Pull request by MichelleXiaoxiaoWang**
Signed-off-by: Xiaoxiao Wang <michellejavajs@gmail.com>

### CODE REVIEW

File: .copyrightignore [hunk 1]
```diff
@@ -0,0 +1,30 @@
+# *******************************************************************************
+# Copyright (c) 2017, 2018 IBM Corp. and others
```
[the remark this answers is not in the document]
Comment (MichelleXiaoxiaoWang) [reply .copyrightignore]: fixed

File: .copyrightignore [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+# *******************************************************************************
+# TODO README ON COPYRIGHT WHITELIST
+# *******************************************************************************/
+/doc/
+#
+/job
+/*.ignore
+debugtools/**/*.ignore
+buildspecs/*.ignore
+/jobs/pipelines/
```
[the remark this answers is not in the document]
Comment (MichelleXiaoxiaoWang) [reply .copyrightignore]: fixed

File: buildenv/jenkins/jobs/infrastructure/copyrightCheck [hunk 3]
```diff
@@ -46,35 +51,70 @@ timeout(time: 6, unit: 'HOURS') {
...
                         returnStdout: true
                     ).trim()
-                    
+
                     // Set a different Copyright regex depending on the Repo the PR is from
-                    if (ghprbGhRepository ==~ "eclipse.*") {
+                    if (ghprbGhRepository ==~ "eclipse.*"){
                         REGEX = "\'Copyright \\(c\\) ([0-9]{4}), ${DATE_YEAR}\'"
-                    } else if (ghprbGhRepository ==~ "ibmruntimes.*") {
+                    } else if (ghprbGhRepository ==~ "ibmruntimes.*"){
```
[the remark this answers is not in the document]
Comment (MichelleXiaoxiaoWang) [reply buildenv/jenkins/jobs/infrastructure/copyrightCheck]: fixed

File: buildenv/jenkins/jobs/infrastructure/copyrightCheck [hunk 4]
```diff
@@ -46,35 +51,70 @@ timeout(time: 6, unit: 'HOURS') {
...
+                             }
+
+                             if (item.contains("#")){
+                                    regexsElm = regexsElm.replaceAll("\\*", "");
+                                    regexsElm = ""; 
+                                }
+                            REGEXLIST << regexsElm ;
+                        }
+                        FILES_LIST.each { fileName ->
+                           REGEXLIST.each { regexses ->
```
[the remark this answers is not in the document]
Comment (MichelleXiaoxiaoWang) [reply buildenv/jenkins/jobs/infrastructure/copyrightCheck]: fixed

File: buildenv/jenkins/jobs/infrastructure/copyrightCheck [hunk 5]
```diff
@@ -46,35 +51,70 @@ timeout(time: 6, unit: 'HOURS') {
...
+                                if (fileName ==~ /${regexses}/ && !WHITELIST_FILES.contains(fileName)) {                                 
+                                    FILES_LIST = FILES_LIST.collect { it.toString() } - "${fileName}";
+                                    echo "Ignoring Files: '${fileName}', appears to be in whiteList";  
+                                    return true
```
[the remark this answers is not in the document]
Comment (MichelleXiaoxiaoWang) [reply buildenv/jenkins/jobs/infrastructure/copyrightCheck]: fixed

File: buildenv/jenkins/jobs/infrastructure/copyrightCheck [hunk 6]
```diff
@@ -46,35 +51,70 @@ timeout(time: 6, unit: 'HOURS') {
...
-                            returnStatus: true)
-                        if(RESULT != 0) {
-                            echo "FAILURE - Copyright date in file: '${it}' appears to be incorrect"
-                            FAIL = true
-                            BAD_FILES << "${it}"
-                        } else {
-                            echo "Copyright date in file: appears to be correct"
+                                    script: "grep -qE ${REGEX} '${file}'",
+                                    returnStatus: true)
+                        if(RESULT != 0){
```
[the remark this answers is not in the document]
Comment (MichelleXiaoxiaoWang) [reply buildenv/jenkins/jobs/infrastructure/copyrightCheck]: fixed

File: buildenv/jenkins/jobs/infrastructure/copyrightCheck [hunk 7]
```diff
@@ -46,35 +51,70 @@ timeout(time: 6, unit: 'HOURS') {
...
-                            script: "grep -qE ${REGEX} '${it}'",
-                            returnStatus: true)
-                        if(RESULT != 0) {
-                            echo "FAILURE - Copyright date in file: '${it}' appears to be incorrect"
-                            FAIL = true
-                            BAD_FILES << "${it}"
-                        } else {
-                            echo "Copyright date in file: appears to be correct"
+                                    script: "grep -qE ${REGEX} '${file}'",
+                                    returnStatus: true)
```
[the remark this answers is not in the document]
Comment (MichelleXiaoxiaoWang) [reply buildenv/jenkins/jobs/infrastructure/copyrightCheck]: the echo is not indented

File: buildenv/jenkins/jobs/infrastructure/copyrightCheck [hunk 8]
```diff
@@ -46,35 +51,70 @@ timeout(time: 6, unit: 'HOURS') {
...
+
+                             if (item.contains("#")){
+                                    regexsElm = regexsElm.replaceAll("\\*", "");
+                                    regexsElm = ""; 
+                                }
+                            REGEXLIST << regexsElm ;
+                        }
+                        FILES_LIST.each { fileName ->
+                           REGEXLIST.each { regexses ->
+                                if (fileName ==~ /${regexses}/ && !WHITELIST_FILES.contains(fileName)) {
```
[the remark this answers is not in the document]
Comment (MichelleXiaoxiaoWang) [reply buildenv/jenkins/jobs/infrastructure/copyrightCheck]: fixed

File: buildenv/jenkins/jobs/infrastructure/copyrightCheck [hunk 9]
```diff
@@ -46,35 +51,70 @@ timeout(time: 6, unit: 'HOURS') {
...
+                        FILES_LIST.each { fileName ->
+                           REGEXLIST.each { regexses ->
+                                if (fileName ==~ /${regexses}/ && !WHITELIST_FILES.contains(fileName)) {                                 
+                                    FILES_LIST = FILES_LIST.collect { it.toString() } - "${fileName}";
+                                    echo "Ignoring Files: '${fileName}', appears to be in whiteList";  
+                                    return true                                                  
+                                }
+                            }
+                        }
+
```
[the remark this answers is not in the document]
Comment (MichelleXiaoxiaoWang) [reply buildenv/jenkins/jobs/infrastructure/copyrightCheck]: fixed

File: buildenv/jenkins/jobs/infrastructure/copyrightCheck [hunk 10]
```diff
@@ -46,35 +51,70 @@ timeout(time: 6, unit: 'HOURS') {
...
+                                if (fileName ==~ /${regexses}/ && !WHITELIST_FILES.contains(fileName)) {                                 
+                                    FILES_LIST = FILES_LIST.collect { it.toString() } - "${fileName}";
+                                    echo "Ignoring Files: '${fileName}', appears to be in whiteList";  
+                                    return true                                                  
+                                }
+                            }
+                        }
+                       
+                    }
+
```
[the remark this answers is not in the document]
Comment (MichelleXiaoxiaoWang) [reply buildenv/jenkins/jobs/infrastructure/copyrightCheck]: fixed

**Comment by MichelleXiaoxiaoWang** [comment]
#**/jobs/**#comment
This will exclude "jobs" path (not root directory) 
screen shot 2018-05-22 at 12 13 27 pm

**Comment by MichelleXiaoxiaoWang** [comment]
1. #**/jobs/**#comment. 
   screen shot 2018-05-22 at 12 13 27 pm
2. /doc
   /*.ignore
   /debugtools/**/*.ignore
   /buildspecs/*.ignore
screen shot 2018-05-22 at 1 29 06 pm
3.*.ignore
screen shot 2018-05-22 at 1 19 33 pm

**Comment by MichelleXiaoxiaoWang** [comment]
Jenkins signed off by check

**Comment by AdamBrousseau** [comment]
While we appreciate the effort put into this PR, we have decided that allowing regexes to ignore blocks of files carries too high of a risk. We would be open to using an ignore file if it were to be a simple list of files to ignore. This would solve the copyright ignore issues without adding risk of accidentally ignoring a file that should have a proper copyright. I would assume given Michelle was an intern she may not be interested in reworking this PR (if I am wrong by all means please post an update). It would be a good first issue for someone who wants to get into Jenkins pipelines. I don't think it would be a large effort to rework this, but I don't know when it would be high enough priority for myself to complete.

**Comment by DanHeidinga** [comment]
@AdamBrousseau should this be closed now?

**Comment by AdamBrousseau** [comment]
Yes this was completed in #5688

**Pull request closed without merging**